### ISSUE TITLE
ComboBox with many items does not scroll to correct value 

### ISSUE THREAD

**Issue by mheicke** [vaadin/flow-components#3470]
### Description of the bug

We have a `ComboBox<ZoneId>` to select the users timezone. As you can imagine, there are a LOT of timezones. Lets say the user selects `Europe/Kiev` and clicks "save", now we store the value for the user. If the user settings screen is opened again, we call `comboBox.setValue(user.getTimeZone())` with the Timezone which we saved before. The TimeZone `Asia/Tbilisi` shows correctly in the ComboBox, but when the dropdown is clicked, the list starts at the top and hasn't scrolled down to `Europe/Kiev` (see screenshot)

<img width="514" alt="Bildschirmfoto 2022-07-18 um 20 14 23" src="https://user-images.githubusercontent.com/246571/179576369-73b4f908-f586-4be6-9c17-2114b659f808.png">

### Expected behavior

I would expect, that if I use comboBox.setValue(xyz), that the element xyz should be scrolled into the visible area. This works on smaller examples, but if you load many items into the ComboBox, it stops to work

### Minimal reproducible example

```
var zoneIds = ZoneId.getAvailableZoneIds().stream()
					.distinct()
					.map(ZoneId::of)
					.collect(Collectors.toList());

timeZone = new ComboBox<>(getTranslation("TimeZone"));
timeZone.setItems(zoneIds);
```

### Versions

- Vaadin / Flow version: 23.1.2
- Java version: 11.0.15
- OS version: MacOS 12.4
- Browser version (if applicable): tested on Safari 15.5

**Comment by knoobie** [comment]
@rolfsmeds instead of scrolling to the correct item, couldn't this be done like you have suggested for the MSCB (showing selected as first; https://github.com/vaadin/web-components/issues/4185)? This would also allow to have the same UX for CB and MSCB as well.

**Comment by mheicke** [comment]
@rolfsmeds thanks for the quick answer. Just as idea: we have the full list - lets say 10.000 items. We have the one item, we want to show (number 1530 of the list). Couldn't the combobox show the items 1530 till 1580 and not load the previous once (just show empty elements here, until they are scrolled into view? I guess that would be a rather complicated fix.

I just checked and there are around 600 timezones - do you think that would be an issue for a page size?

**Comment by TatuLund** [comment]
> I just checked and there are around 600 timezones - do you think that would be an issue for a page size?

Yes, I think so. There is limit of 500 items for safety reasons.

**Comment by rolfsmeds** [comment]
> couldn't this be done like you have suggested for the MSCB (showing selected as first; [vaadin/web-components#4185](https://github.com/vaadin/web-components/issues/4185))? This would also allow to have the same UX for CB and MSCB as well.

We're considering that, yes, so although that wouldn't really _solve_ the problem per se, it would at least let you find the selected item easily, thus circumventing the problem.

> Couldn't the combobox show the items 1530 till 1580 and not load the previous once

I'm not quite sure, but I suspect we'd have the same problem there again: there's no way for the CB to know _where_ in the data the selected item is, so it can't know which page to fetch.

**Comment by TatuLund** [comment]
One option to workaround is to use Light ComboBox add-on. This component is based on the same web component as the regular ComboBox with the difference that it skips the lazy loading mechanism. Hence scroll to item works. However this component is not suitable for large data sets.

https://vaadin.com/directory/component/light-combobox

**Comment by web-padawan** [comment]
This was covered by `ComboBox.setFocusSelectedItem(boolean)` opt-in API added in 25.2 - see #9239. Closing.
